lib/targets.go

```diff
@@ -131,7 +131,10 @@ func NewJSONTargeter(src io.Reader, body []byte, header http.Header) Targeter {
 		defer dec.Unlock()
 
 		var t Target
-		if err = dec.Decode(&t); err != nil && err != io.EOF {
+		if err = dec.Decode(&t); err != nil {
+			if err == io.EOF {
+				err = ErrNoTargets
```

Comment:
I think this would work:

```go
func NewJSONTargeter(src io.Reader, body []byte, header http.Header) Targeter {
	type reader struct {
		*bufio.Reader
		sync.Mutex
	}
	rd := reader{Reader: bufio.NewReader(src)}

	return func(tgt *Target) (err error) {
		if tgt == nil {
			return ErrNilTarget
		}

		rd.Lock()
		defer rd.Unlock()

		var line []byte
		for len(line) == 0 {
			if line, err = rd.ReadBytes('\n'); err != nil {
				break
			}
			line = bytes.TrimSpace(line) // Skip empty lines
		}

		if err != nil {
			if err == io.EOF {
				err = ErrNoTargets
			}
			return err
		}

		var t Target
		if err = json.Unmarshal(line, &t); err != nil {
			return err
		} else if t.Method == "" {
			return ErrNoMethod
		} else if t.URL == "" {
			return ErrNoURL
		}

		tgt.Method = t.Method
		tgt.URL = t.URL
		if tgt.Body = body; len(t.Body) > 0 {
			tgt.Body = t.Body
		}

		if tgt.Header == nil {
			tgt.Header = http.Header{}
		}

		for k, vs := range header {
			tgt.Header[k] = append(tgt.Header[k], vs...)
		}

		for k, vs := range t.Header {
			tgt.Header[k] = append(tgt.Header[k], vs...)
		}

		return nil
	}
}
```

Reply:
True, this will now cause an error when decoding multiple lines without the `\n` though.

Reply:
Sorry, I didnt see any of your comments until 5 minutes ago